backend/src/middlewares/auth/strategies/google.strategy.ts

```diff
@@ -0,0 +1,75 @@
+import { ENV } from '@common/enums/app/env.enum';
+import { prisma } from '@data/prisma-client';
+import { verifyToken } from '@helpers/helpers';
+import { JwtPayload } from 'jsonwebtoken';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+
+const googleStrategy = new GoogleStrategy(
+  {
+    clientID: ENV.GOOGLE.CLIENT_ID as string,
+    clientSecret: ENV.GOOGLE.CLIENT_SECRET as string,
+    callbackURL: `${ENV.API.V1_PREFIX}/auth/google/redirect`,
+    passReqToCallback: true,
+    scope: ['profile', 'email'],
+  },
+  async (req, accessToken, refreshToken, profile, done) => {
+    let user;
+    try {
+      user = await prisma.user.findFirst({
+        where: {
+          User_Security: {
+            google_acc_id: profile.id,
+          },
+        },
+      });
+      if (!user) {
+        const payload =
```

Comment:
Please use guard clauses instead of having redundant if-else nesting -https://learningactors.com/javascript-guard-clauses-how-you-can-refactor-conditional-logic/